lib/orderbook/OrderBook.ts

```diff
@@ -207,27 +220,42 @@ class OrderBook extends EventEmitter {
     }
 
     this.pairs.delete(pairId);
-    this.matchingEngines.delete(pairId);
+    this.tradingPairs.delete(pairId);
 
     if (this.pool) {
       this.pool.updateHandshake({ pairs: this.pairIds });
     }
     return pair.destroy();
   }
 
-  public addLimitOrder = async (order: orders.OwnOrder, onUpdate?: (e: PlaceOrderEvent) => void): Promise<PlaceOrderResult> => {
+  public placeLimitOrder = async (order: orders.OwnOrder, onUpdate?: (e: PlaceOrderEvent) => void): Promise<PlaceOrderResult> => {
     const stampedOrder = this.stampOwnOrder(order);
-    return this.addOwnOrder(stampedOrder, false, onUpdate, Date.now() + OrderBook.MAX_ADD_OWN_ORDER_ITERATIONS_TIME);
+    if (this.nomatching) {
+      this.addOwnOrder(stampedOrder);
+      onUpdate && onUpdate({ case: PlaceOrderEventCase.RemainingOrder, payload: stampedOrder });
+
+      return Promise.resolve({
```

Comment:
I personally think it'd be ok - since anytime this method is called it would still need an `await` or `then` which would end up with a synchronous `PlaceOrderResult` value - but I think it's fine too to keep the method async and drop the `Promise.resolve`. Whichever you prefer.

Reply:
I actually thought that it would be an error to do `await` on non-promise, but apparently JS protects against it by wrapping the result in `Promise.resolve`. Anyway, I ended up just removing the `async` keyword.